Give tidy the good news about C-SKY

workingjubilee opened the pull request:
It seems this was overlooked in https://github.com/rust-lang/rust/pull/125472 because we don't test C-SKY much yet.

Fixes #125697

r? @erikdesjardins

workingjubilee committed:
give tidy the good news about C-SKY

workingjubilee commented:
:ferrisCluelesser:

r? bootstrap

Kobzol commented:
r? @Kobzol

@bors r+

bors commented:
:pushpin: Commit 084f0583adff6a835a0fc913490099e4a19493ae has been approved by `Kobzol`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol 084f0583adff6a835a0fc913490099e4a19493ae -->
<!-- homu: {"type":"Approved","sha":"084f0583adff6a835a0fc913490099e4a19493ae","approver":"Kobzol","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

lqd commented:
Can the test be re-enabled? https://github.com/rust-lang/rust/blob/751691271d76b8435559200b84d1947c2bd735bd/tests/ui/abi/compatibility.rs#L61-L65

RalfJung commented:
@bors r-

workingjubilee commented:
I believe the problem with CSKY was that when we supported sufficiently early-enough versions of LLVM, those ones didn't have the actual component in them.

workingjubilee commented:
Since the change to LLVM 18 and the dropping of an old LLVM version have happened since last November, the logfiles would have been misinforming you anyways.

workingjubilee committed:
tests: reenable ABI compatibility test for csky

workingjubilee commented:
It passes locally, and I was able to make the test emit a compile error for specifically the csky revision, so it is in fact building it.

RalfJung commented:
Yeah the test already passed locally back then. But then somehow on CI the test failed.

> I believe the problem with CSKY was that when we supported sufficiently early-enough versions of LLVM, those ones didn't have the actual component in them.

That doesn't explain it though -- the test had the needs-component line, so if the LLVM on CI does not have csky it should have skipped the test. Maybe our implementation of needs-component is buggy...

workingjubilee commented:
>  Maybe our implementation of needs-component is buggy...

do you really think people would really do that? just go out on the internet and write bugs in their test code? :^)

lqd commented:
LLVM component shenanigans and unclear CI requirements could mess a rollup, @bors r=Kobzol,lqd rollup=iffy

bors commented:
:pushpin: Commit 81bc4d01f9fee53918038c13ae0204788dc4e760 has been approved by `Kobzol,lqd`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol,lqd 81bc4d01f9fee53918038c13ae0204788dc4e760 -->
<!-- homu: {"type":"Approved","sha":"81bc4d01f9fee53918038c13ae0204788dc4e760","approver":"Kobzol,lqd","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Kobzol commented:
There was some weird csky behavior in https://github.com/rust-lang/rust/pull/125141. The `x86_64-gnu-llvm-17` job was outputting
```
error: could not create LLVM TargetMachine for triple: csky-unknown-linux-gnuabiv2: No available targets are compatible with triple "csky-unknown-linux-gnuabiv2"
```
in `tests/run-make/print-target-list`. The test was obviously succeeding before this PR, but somehow unrelated changes from the PR somehow also broke csky (?). Or maybe the error is swallowed because of some test bug :shrug:

lqd commented:
We'll see what the PR builder says about this one.

We could also take a look at CI's record of tests results with their ignore reason, if we have it until that far back, and check for csky there.

bors commented:
:hourglass: Testing commit 81bc4d01f9fee53918038c13ae0204788dc4e760 with merge 0e859e47c7bed01f205b45f49a22d16e38e4b827...
<!-- homu: {"type":"BuildStarted","head_sha":"81bc4d01f9fee53918038c13ae0204788dc4e760","merge_sha":"0e859e47c7bed01f205b45f49a22d16e38e4b827"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9282739189/job/25541433473) [(plain)](https://github.com/rust-lang-ci/rust/commit/0e859e47c7bed01f205b45f49a22d16e38e4b827/checks/25541433473/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   2: compiletest::header::ignore_llvm
   3: compiletest::header::make_test_description::<std::fs::File>::{closure#0}
   4: compiletest::header::iter_header::<std::fs::File>
   5: compiletest::header::make_test_description::<std::fs::File>
   6: <alloc::vec::into_iter::IntoIter<core::option::Option<&str>> as core::iter::traits::iterator::Iterator>::fold::<(), core::iter::adapters::map::map_fold<core::option::Option<&str>, test::types::TestDescAndFn, (), compiletest::make_test::{closure#1}, core::iter::traits::iterator::Iterator::for_each::call<test::types::TestDescAndFn, <alloc::vec::Vec<test::types::TestDescAndFn>>::extend_trusted<core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<core::option::Option<&str>>, compiletest::make_test::{closure#1}>>::{closure#0}>::{closure#0}>::{closure#0}>
   7: <core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<core::option::Option<&str>>, compiletest::make_test::{closure#1}> as core::iter::traits::iterator::Iterator>::fold::<(), core::iter::traits::iterator::Iterator::for_each::call<test::types::TestDescAndFn, <alloc::vec::Vec<test::types::TestDescAndFn>>::extend_trusted<core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<core::option::Option<&str>>, compiletest::make_test::{closure#1}>>::{closure#0}>::{closure#0}>
   8: <alloc::vec::Vec<test::types::TestDescAndFn> as alloc::vec::spec_from_iter::SpecFromIter<test::types::TestDescAndFn, core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<core::option::Option<&str>>, compiletest::make_test::{closure#1}>>>::from_iter
  10: compiletest::collect_tests_from_dir
  11: compiletest::collect_tests_from_dir
  12: compiletest::make_tests
  13: compiletest::run_tests
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/9282739189/job/25542004327)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/9282739189/job/25542004327","builder_name":"checks-actions"} -->

lqd commented:
heh

> missing LLVM component: csky

RalfJung commented:
Oh, so on CI we actually ignore the `needs-llvm-component` part and run the test anyway?
That should be documented and mentioned in the error message, it is very confusing.
But it is good to know that we actually enforce that the test runs somewhere.  If we unset COMPILETEST_NEEDS_ALL_LLVM_COMPONENTS, will the test be run anywhere at all?

nikic commented:
We don't run the test, but we check that all components are present in CI test runs. And yes, the test will get run on builders using our own LLVM. But we don't control which targets Ubuntu's LLVM includes, so we shouldn't use the option there.

RalfJung commented:
> We don't run the test, but we check that all components are present in CI test runs.

If the component is present, then we also run the test, no?
Doesn't this mean the new check in https://github.com/rust-lang/rust/pull/125472 is unnecessary, we would already have noticed when someone adds a non-existing component?

https://github.com/rust-lang/rust/pull/125710 should help clarify the error we are seeing here.

nikic commented:
> Doesn't this mean the new check in #125472 is unnecessary, we would already have noticed when someone adds a non-existing component?

Yes-ish. I believe CI would fail, but having it in tidy is probably still a nice convenience to catch the issue during local development and PR CI already.

> Ah so there are still other runners where COMPILETEST_NEEDS_ALL_LLVM_COMPONENTS is set?

See https://github.com/rust-lang/rust/blob/5870f1ccbbb2661c92cdc356231386210a64d723/src/ci/run.sh#L172-L176. It's currently set unless `IS_NOT_LATEST_LLVM` which is only set for the llvm-17 job. It's enabled everywhere else.

bors commented:
:umbrella: The latest upstream changes (presumably #125695) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

workingjubilee commented:
I am simply going to move the logic under `$EXTERNAL_LLVM`, since the answer to "does this LLVM have the components we need?" is "well, did we _build it with them_?"

workingjubilee committed:
ci: Consolidate $IS_NOT_LATEST_LLVM into $EXTERNAL_LLVM

We want to only demand that we check for all components we expect
if we actually built the components we expect, which means
we built the LLVM. Otherwise, it isn't worth checking.

nikic commented:
@bors r+

bors commented:
:pushpin: Commit 60c30f57c1c4937f73416bf06295295076aa0c15 has been approved by `nikic`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=nikic 60c30f57c1c4937f73416bf06295295076aa0c15 -->
<!-- homu: {"type":"Approved","sha":"60c30f57c1c4937f73416bf06295295076aa0c15","approver":"nikic","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 60c30f57c1c4937f73416bf06295295076aa0c15 with merge 23ea77b8edc902f4a90cda62af66f8b300e5de54...
<!-- homu: {"type":"BuildStarted","head_sha":"60c30f57c1c4937f73416bf06295295076aa0c15","merge_sha":"23ea77b8edc902f4a90cda62af66f8b300e5de54"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/9294015105/job/25581591858)
Approved by: nikic
Pushing 23ea77b8edc902f4a90cda62af66f8b300e5de54 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"nikic","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/9294015105/job/25581591858"},"merge_sha":"23ea77b8edc902f4a90cda62af66f8b300e5de54"} -->

rust-timer commented:
Finished benchmarking commit (23ea77b8edc902f4a90cda62af66f8b300e5de54): [comparison URL](https://perf.rust-lang.org/compare.html?start=debd22da66cfa97c74040ebf68e420672ac8560e&end=23ea77b8edc902f4a90cda62af66f8b300e5de54&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=debd22da66cfa97c74040ebf68e420672ac8560e&end=23ea77b8edc902f4a90cda62af66f8b300e5de54&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=debd22da66cfa97c74040ebf68e420672ac8560e&end=23ea77b8edc902f4a90cda62af66f8b300e5de54&stat=max-rss)
<details>
<summary>Results (primary -0.5%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 3.8%  | [2.2%, 5.4%]   | 2     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -2.2% | [-3.1%, -1.4%] | 5     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -0.5% | [-3.1%, 5.4%]  | 7     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=debd22da66cfa97c74040ebf68e420672ac8560e&end=23ea77b8edc902f4a90cda62af66f8b300e5de54&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=debd22da66cfa97c74040ebf68e420672ac8560e&end=23ea77b8edc902f4a90cda62af66f8b300e5de54&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 666.699s -> 669.162s (0.37%)
**Artifact size**: 318.90 MiB -> 318.80 MiB (-0.03%)

erikdesjardins commented:
Hmmm, I didn't realize that this was already caught in (non-PR) CI.

I'm inclined to just revert the original PR. It's a rare enough issue that it just needs to be caught _somehow_, and it shouldn't be a huge issue that it takes a bors run to do so.
